### PR TITLE
48 google footfall

### LINKED ISSUE #48: Collect and process Google footfall data

### PR DESCRIPTION
**Pull request by Juan-Mateos**
Closes #48 

For now it only extracts country-level data (sub-regional entries are labelled with ISO 3166 codes, and I couldn't find a lookup between those and ONS geography codes. 😢 )

### CODE REVIEW

File: sg_covid_impact/google_footfall.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    c_long = c.dropna(axis=1).melt(
+        id_vars=["country_region", "country_region_code", "date"]
+    )
+    c_long.to_csv(
+        f"{project_dir}/data/processed/google_footfall_{country_code}.csv", index=False
+    )
+
+
+def load_uk_footfall():
+    return pd.read_csv(f"{project_dir}/data/processed/google_footfall_gb.csv")
```
Comment (bishax) [inline sg_covid_impact/google_footfall.py]: `gb` -> `GB`

Comment (bishax) [reply sg_covid_impact/google_footfall.py]: Docstring

File: sg_covid_impact/google_footfall.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
+import requests
```
Comment (bishax) [inline sg_covid_impact/google_footfall.py:3]: Docstring, e.g. """Fetch, clean, and load google footfall data (<link to webpage>)."""

File: sg_covid_impact/google_footfall.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    # Only keep the rows with subregion field missing (ie the national data)
+    c = c.loc[c["sub_region_1"].isna()]
+    
+    # Make tidy
+    c_long = c.dropna(axis=1).melt(
+        id_vars=["country_region", "country_region_code", "date"]
+    )
+    c_long.to_csv(
+        f"{project_dir}/data/processed/google_footfall_{country_code}.csv", index=False
+    )
```
Comment (bishax) [inline sg_covid_impact/google_footfall.py]: Cleaner to return the data and then save it elsewhere. Processing should be decoupled from saving.

File: sg_covid_impact/google_footfall.py [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+
+def load_uk_footfall():
+    return pd.read_csv(f"{project_dir}/data/processed/google_footfall_gb.csv")
+
+
+if __name__ == "__main__":
+    if not os.path.exists(_TARGET_DIR):
```
Comment (bishax) [inline sg_covid_impact/google_footfall.py:53]: This might trip us up if we want to update

File: sg_covid_impact/google_footfall.py [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+        os.makedirs(_TARGET_DIR, exist_ok=True)
+        fetch_footfall_data()
+        process_footfall_data("gb")
```
Comment (bishax) [inline sg_covid_impact/google_footfall.py]: `GB`

File: sg_covid_impact/google_footfall.py [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+    reg_g = requests.get(_GOOG_URL)
+    z = ZipFile(BytesIO(reg_g.content))
+    z.extractall(_TARGET_DIR)
+
+
+def process_footfall_data(country_code):
+    """Reads and processes country data for a country
+    Args:
+        country_code (str): country iso code
+        TODO: adapt to extract subregional data!
```
Comment (bishax) [inline sg_covid_impact/google_footfall.py:30]: Is this required to fulfill the PR?

**Review by bishax — COMMENTED** [review]
A small fix for the country code and a few minor changes then GTM.

File: sg_covid_impact/google_footfall.py (hunk 6 above)
Comment (Juan-Mateos) [reply sg_covid_impact/google_footfall.py:30]: No - for now we are just focusing on national trends + sub-national data only seems to be available for a subset of LADs. I would create a new issue for it.

**Pull request merged**